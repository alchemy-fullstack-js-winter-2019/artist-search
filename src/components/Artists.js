import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default function Artists({ artists }) {
  const artistList = artists.map(artist => {
    return (
      <Link to={`/artist/${artist.id}`} key={artist.id}>
        <li key={artist.id}>{artist.name}</li>
      </Link>
    );
  });
  return (
    <ul>{artistList}</ul>
  ); 
}

Artists.propTypes = {
  artists: PropTypes.array.isRequired
}; 
