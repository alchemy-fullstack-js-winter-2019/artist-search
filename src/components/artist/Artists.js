import React from 'react';
import PropTypes from 'prop-types';
import { withPaging } from '../paging/Paging';

function Artists({ artists }) {
  const { listOfArtists } = artists;
  return (
    
  );
}

Artists.propTypes = { 
  artists: PropTypes.array.isRequired
};

export default withPaging(Artists);

