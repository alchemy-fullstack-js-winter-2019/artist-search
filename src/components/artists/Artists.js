// import React, { PureComponent } from 'react';
// import PropTypes from 'prop-types';
// import Artist from './Artist';
// import { getArtists } from '../../services/artistsApi';

// export class Artists extends PureComponent {
//   static propTypes = {
//     results: PropTypes.array.isRequired
//   };
//   render() {
//     const artists = this.props.results.map(artist => {
//       return (
//         <Artist key={artist.id} artist={artist}/>
//       );
//     });
//     return (
//       <ul>
//         {artists}
//       </ul>
//     );
//   }
// }

// Artists.propTypes = {
//   artist: PropTypes.array
// };

// export default Artists;
