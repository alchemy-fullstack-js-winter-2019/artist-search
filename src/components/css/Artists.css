.artists {
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
  li {
    padding: 5px;
    list-style: none;
  }
  li:hover {
    background: rgba(255, 243, 76, 0.781)
  }
  ul {
    padding-left: 0;
  }
}
