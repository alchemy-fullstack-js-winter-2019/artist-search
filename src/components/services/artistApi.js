export const getArtists = ({ page, search }) => {
  return fetch(`http://musicbrainz.org/ws/2/artist?query=${search}&fmt=json&offset=${page}`, {
    headers: {
      'origin': null
    }
  })
    .then(res => res.json())
    .then(json => ({
      totalPages: Math.ceil(json.count / 25),
      results: json.artists
    }));
};

export const getArtistDetail = (id) => {
  return fetch(`http://musicbrainz.org/ws/2/artist/${id}?fmt=json&inc=works`, {
    headers: {
      'origin': null
    }
  })
    .then(res => res.json())
    .then(json => ({
      results: json.works
    }));
};

