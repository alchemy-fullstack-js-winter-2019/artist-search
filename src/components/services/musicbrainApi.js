export const getArtist = (artist) => {
    return fetch(`http://musicbrainz.org/ws/2/artist?query=${artist}&fmt=json&limit=25`)
        .then(res => {
            return res.json();
        });
};
export const getWorks = (id) => {
    return fetch(`http://musicbrainz.org/ws/2/artist/${id}?fmt=json&inc=works`)
        .then(res => {
            return res.json();
        });
};

