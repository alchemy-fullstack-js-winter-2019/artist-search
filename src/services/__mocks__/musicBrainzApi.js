const artists = {
  'created': '2019-02-22T18:41:06.449Z',
  'count': 33,
  'offset': 0,
  'artists': [
    {
      'id': '57f73e51-5b95-485b-9102-bc7bd23584f7',
      'type': 'Group',
      'type-id': 'e431f5f6-b5d2-343d-8b36-72607fffb74b',
      'score': 100,
      'name': 'Mumford’s',
      'sort-name': 'Mumford’s',
      'life-span': {
        'begin': '2008',
        'ended': null
      },
      'tags': [
        {
          'count': 1,
          'name': 'iowa'
        }
      ]
    },
    {
      'id': 'e5379ae1-cb4c-46dc-89f4-3d22791f589a',
      'score': 86,
      'name': 'Quincy Mumford',
      'sort-name': 'Mumford, Quincy',
      'life-span': {
        'ended': null
      }
    },
    {
      'id': '2f3613cf-568c-4e18-ba59-ce6d50f05f00',
      'type': 'Group',
      'type-id': 'e431f5f6-b5d2-343d-8b36-72607fffb74b',
      'score': 85,
      'name': 'Brock Mumford',
      'sort-name': 'Brock Mumford',
      'life-span': {
        'ended': null
      }
    },
    {
      'id': '712756f8-2e2e-4c2f-80a2-07e239b1a0fe',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 85,
      'name': 'Brenton Mumford',
      'sort-name': 'Mumford, Brenton',
      'life-span': {
        'ended': null
      }
    },
    {
      'id': 'f1800bcb-60e3-48d0-9308-413a6a56de83',
      'score': 85,
      'name': 'Marvin Mumford',
      'sort-name': 'Marvin Mumford',
      'life-span': {
        'ended': null
      }
    },
    {
      'id': 'b245d88e-b04f-4748-8937-2f2581a84437',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 85,
      'name': 'Tamara Mumford',
      'sort-name': 'Mumford, Tamara',
      'gender': 'female',
      'country': 'US',
      'area': {
        'id': '489ce91b-6658-3307-9877-795b68554c98',
        'type': 'Country',
        'type-id': '06dd0ae4-8c74-30bb-b43d-95dcedf961de',
        'name': 'United States',
        'sort-name': 'United States',
        'life-span': {
          'ended': null
        }
      },
      'disambiguation': 'mezzo soprano',
      'life-span': {
        'ended': null
      }
    },
    {
      'id': 'f823c878-116d-4a42-998f-440617d5c990',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Jeffrey Mumford',
      'sort-name': 'Mumford, Jeffrey',
      'gender': 'male',
      'country': 'US',
      'area': {
        'id': '489ce91b-6658-3307-9877-795b68554c98',
        'type': 'Country',
        'type-id': '06dd0ae4-8c74-30bb-b43d-95dcedf961de',
        'name': 'United States',
        'sort-name': 'United States',
        'life-span': {
          'ended': null
        }
      },
      'life-span': {
        'begin': '1955',
        'ended': null
      }
    },
    {
      'id': '30415593-f831-42a1-ba3f-fbaef1272815',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Marcus Mumford',
      'sort-name': 'Mumford, Marcus',
      'gender': 'male',
      'country': 'GB',
      'area': {
        'id': '8a754a16-0027-3a29-b6d7-2b40ea0481ed',
        'type': 'Country',
        'type-id': '06dd0ae4-8c74-30bb-b43d-95dcedf961de',
        'name': 'United Kingdom',
        'sort-name': 'United Kingdom',
        'life-span': {
          'ended': null
        }
      },
      'begin-area': {
        'id': 'bf51dede-2c6c-4908-abaa-3d6b491d0cf5',
        'type': 'City',
        'type-id': '6fd8f29a-3d0a-32fc-980d-ea697b69da78',
        'name': 'Orange',
        'sort-name': 'Orange',
        'life-span': {
          'ended': null
        }
      },
      'ipis': [
        '00558912024'
      ],
      'life-span': {
        'begin': '1987-01-31',
        'ended': null
      },
      'aliases': [
        {
          'sort-name': 'Mumford, Marcus Oliver Johnstone',
          'type-id': 'd4dcd0c0-b341-3612-a332-c0ce797b25cf',
          'name': 'Marcus Oliver Johnstone Mumford',
          'locale': null,
          'type': 'Legal name',
          'primary': null,
          'begin-date': null,
          'end-date': null
        }
      ]
    },
    {
      'id': '12500691-034d-46b8-9831-293bd4244b07',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Craig Mumford',
      'sort-name': 'Mumford, Craig',
      'life-span': {
        'ended': null
      },
      'tags': [
        {
          'count': 1,
          'name': 'production music'
        }
      ]
    },
    {
      'id': 'c44e9c22-ef82-4a77-9bcd-af6c958446d6',
      'type': 'Group',
      'type-id': 'e431f5f6-b5d2-343d-8b36-72607fffb74b',
      'score': 84,
      'name': 'Mumford & Sons',
      'sort-name': 'Mumford & Sons',
      'country': 'GB',
      'area': {
        'id': '8a754a16-0027-3a29-b6d7-2b40ea0481ed',
        'type': 'Country',
        'type-id': '06dd0ae4-8c74-30bb-b43d-95dcedf961de',
        'name': 'United Kingdom',
        'sort-name': 'United Kingdom',
        'life-span': {
          'ended': null
        }
      },
      'begin-area': {
        'id': 'f03d09b3-39dc-4083-afd6-159e3f0d462f',
        'type': 'City',
        'type-id': '6fd8f29a-3d0a-32fc-980d-ea697b69da78',
        'name': 'London',
        'sort-name': 'London',
        'life-span': {
          'ended': null
        }
      },
      'life-span': {
        'begin': '2007-12',
        'ended': null
      },
      'aliases': [
        {
          'sort-name': 'Mumford and Sons',
          'type-id': '1937e404-b981-3cb7-8151-4c86ebfc8d8e',
          'name': 'Mumford and Sons',
          'locale': null,
          'type': 'Search hint',
          'primary': null,
          'begin-date': null,
          'end-date': null
        }
      ],
      'tags': [
        {
          'count': 1,
          'name': 'rock'
        },
        {
          'count': 1,
          'name': 'alternative rock'
        },
        {
          'count': 1,
          'name': 'americana'
        },
        {
          'count': 1,
          'name': 'alt-country'
        },
        {
          'count': 1,
          'name': 'indie folk'
        },
        {
          'count': 1,
          'name': 'pop rock'
        },
        {
          'count': 2,
          'name': 'folk pop'
        },
        {
          'count': 0,
          'name': 'remember'
        }
      ]
    },
    {
      'id': '98bca344-cede-4bb8-bdf1-e768db65e809',
      'score': 84,
      'name': 'Joy Mumford',
      'sort-name': 'Mumford, Joy',
      'life-span': {
        'ended': null
      }
    },
    {
      'id': '8b7fa1af-67f6-4c6a-82e3-0bb3b122aa89',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Thad Mumford',
      'sort-name': 'Mumford, Thad',
      'life-span': {
        'ended': null
      }
    },
    {
      'id': '90ce8fa8-999f-4269-9024-b3c8d0861c6f',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Amy Mumford',
      'sort-name': 'Mumford, Amy',
      'life-span': {
        'ended': null
      }
    },
    {
      'id': '2e6d9dd1-72cb-48a1-83ba-fcc162ab69dd',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Jarrett Mumford',
      'sort-name': 'Mumford, Jarrett',
      'life-span': {
        'ended': null
      }
    },
    {
      'id': 'ab6daaef-5532-4431-ad41-f858a8ea59a4',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Todd Mumford',
      'sort-name': 'Mumford, Todd',
      'gender': 'male',
      'country': 'US',
      'area': {
        'id': '489ce91b-6658-3307-9877-795b68554c98',
        'type': 'Country',
        'type-id': '06dd0ae4-8c74-30bb-b43d-95dcedf961de',
        'name': 'United States',
        'sort-name': 'United States',
        'life-span': {
          'ended': null
        }
      },
      'ipis': [
        '00450315199'
      ],
      'life-span': {
        'ended': null
      }
    },
    {
      'id': 'af26be7a-5bbe-4413-849b-b23ae96e5a46',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Jill Mumford',
      'sort-name': 'Mumford, Jill',
      'life-span': {
        'ended': null
      }
    },
    {
      'id': '3c63ac9b-3df0-480c-bf97-0ef745ab2d2a',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'John Mumford',
      'sort-name': 'Mumford, John',
      'gender': 'male',
      'life-span': {
        'ended': null
      }
    },
    {
      'id': '9ba4a805-edde-41ba-b625-aeaa7e106553',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Myles Mumford',
      'sort-name': 'Mumford, Myles',
      'gender': 'male',
      'life-span': {
        'ended': null
      }
    },
    {
      'id': '972a0286-6297-4b68-9c15-f8a2cf2a6a94',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Daniel Mumford',
      'sort-name': 'Mumford, Daniel',
      'gender': 'male',
      'life-span': {
        'ended': null
      }
    },
    {
      'id': '44497310-aa8c-43b5-95da-8474793aa1f4',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Kate Mumford',
      'sort-name': 'Mumford, Kate',
      'gender': 'female',
      'country': 'GB',
      'area': {
        'id': '8a754a16-0027-3a29-b6d7-2b40ea0481ed',
        'type': 'Country',
        'type-id': '06dd0ae4-8c74-30bb-b43d-95dcedf961de',
        'name': 'United Kingdom',
        'sort-name': 'United Kingdom',
        'life-span': {
          'ended': null
        }
      },
      'life-span': {
        'ended': null
      }
    },
    {
      'id': 'a7f3c7f3-0d5f-4cdb-9095-8c71f5c70c3a',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Mark Mumford',
      'sort-name': 'Mumford, Mark',
      'gender': 'male',
      'life-span': {
        'ended': null
      }
    },
    {
      'id': '88911794-21db-4711-8d19-1818a56a7ba3',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Mike Mumford',
      'sort-name': 'Mumford, Mike',
      'gender': 'male',
      'country': 'US',
      'area': {
        'id': '489ce91b-6658-3307-9877-795b68554c98',
        'type': 'Country',
        'type-id': '06dd0ae4-8c74-30bb-b43d-95dcedf961de',
        'name': 'United States',
        'sort-name': 'United States',
        'life-span': {
          'ended': null
        }
      },
      'life-span': {
        'ended': null
      }
    },
    {
      'id': 'd2930f5d-afd4-4a9f-b176-a564dda977b9',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Don Mumford',
      'sort-name': 'Mumford, Don',
      'gender': 'male',
      'country': 'US',
      'area': {
        'id': '489ce91b-6658-3307-9877-795b68554c98',
        'type': 'Country',
        'type-id': '06dd0ae4-8c74-30bb-b43d-95dcedf961de',
        'name': 'United States',
        'sort-name': 'United States',
        'life-span': {
          'ended': null
        }
      },
      'begin-area': {
        'id': '7c83a6c9-4e5b-4993-9c11-872bcdfe693d',
        'type': 'City',
        'type-id': '6fd8f29a-3d0a-32fc-980d-ea697b69da78',
        'name': 'Lawrence',
        'sort-name': 'Lawrence',
        'life-span': {
          'ended': null
        }
      },
      'end-area': {
        'id': '2bda1647-f64b-4277-acce-e9c1296d065e',
        'type': 'City',
        'type-id': '6fd8f29a-3d0a-32fc-980d-ea697b69da78',
        'name': 'Des Moines',
        'sort-name': 'Des Moines',
        'life-span': {
          'ended': null
        }
      },
      'life-span': {
        'begin': '1954-02-09',
        'end': '2007-07-06',
        'ended': true
      },
      'tags': [
        {
          'count': 1,
          'name': 'jazz'
        }
      ]
    },
    {
      'id': 'ef12e8ad-b88c-4cfc-8dce-2d6df5705d2e',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Eugene Mumford',
      'sort-name': 'Mumford, Eugene',
      'gender': 'male',
      'country': 'US',
      'area': {
        'id': '489ce91b-6658-3307-9877-795b68554c98',
        'type': 'Country',
        'type-id': '06dd0ae4-8c74-30bb-b43d-95dcedf961de',
        'name': 'United States',
        'sort-name': 'United States',
        'life-span': {
          'ended': null
        }
      },
      'life-span': {
        'begin': '1925-06-24',
        'end': '1977-05-29',
        'ended': true
      },
      'aliases': [
        {
          'sort-name': 'Gene Mumford',
          'name': 'Gene Mumford',
          'locale': null,
          'type': null,
          'primary': null,
          'begin-date': null,
          'end-date': null
        }
      ]
    },
    {
      'id': '60300959-f2a2-4abd-b964-40cc86680565',
      'type': 'Person',
      'type-id': 'b6e035f4-3ce9-331c-97df-83397230b0df',
      'score': 84,
      'name': 'Paul Mumford',
      'sort-name': 'Mumford Paul',
      'gender': 'male',
      'country': 'US',
      'area': {
        'id': '489ce91b-6658-3307-9877-795b68554c98',
        'type': 'Country',
        'type-id': '06dd0ae4-8c74-30bb-b43d-95dcedf961de',
        'name': 'United States',
        'sort-name': 'United States',
        'life-span': {
          'ended': null
        }
      },
      'life-span': {
        'ended': null
      }
    }
  ]
};

const artist = {
  'gender-id': null,
  'type': 'Group',
  'gender': null,
  'name': 'Mumford & Sons',
  'type-id': 'e431f5f6-b5d2-343d-8b36-72607fffb74b',
  'sort-name': 'Mumford & Sons',
  'begin_area': {
    'disambiguation': '',
    'name': 'London',
    'sort-name': 'London',
    'id': 'f03d09b3-39dc-4083-afd6-159e3f0d462f'
  },
  'end_area': null,
  'disambiguation': '',
  'life-span': {
    'begin': '2007-12',
    'ended': false,
    'end': null
  },
  'id': 'c44e9c22-ef82-4a77-9bcd-af6c958446d6',
  'isnis': [
    '0000000121955760'
  ],
  'works': [
    {
      'type-id': null,
      'languages': [
        'eng'
      ],
      'iswcs': [],
      'id': '15dc1e5d-7aff-488e-a6ff-40ba51f326b5',
      'disambiguation': '',
      'attributes': [],
      'type': null,
      'language': 'eng',
      'title': 'Beloved'
    },
    {
      'language': 'eng',
      'disambiguation': '',
      'attributes': [],
      'type': null,
      'languages': [
        'eng'
      ],
      'type-id': null,
      'iswcs': [],
      'id': '1659994e-8632-4b0d-ac5c-11d1f7b85dbe',
      'title': 'Darkness Visible'
    },
    {
      'title': 'After the Storm',
      'type': null,
      'disambiguation': '',
      'attributes': [],
      'language': 'eng',
      'id': '20f0969b-9c25-4760-9abc-395b2a1a8763',
      'type-id': null,
      'iswcs': [
        'T-905.465.280-1'
      ],
      'languages': [
        'eng'
      ]
    },
    {
      'title': 'Holland Road',
      'disambiguation': '',
      'attributes': [],
      'type': null,
      'language': 'eng',
      'iswcs': [
        'T-909.234.735-8'
      ],
      'type-id': null,
      'languages': [
        'eng'
      ],
      'id': '2466b22f-d55c-42a7-83b1-664ae819f753'
    },
    {
      'id': '2ce87533-900f-4c8f-bf8c-20b1506a17cc',
      'type-id': null,
      'iswcs': [
        'T-909.234.729-0'
      ],
      'languages': [
        'eng'
      ],
      'language': 'eng',
      'type': null,
      'disambiguation': '',
      'attributes': [],
      'title': 'Babel'
    },
    {
      'title': 'Delta',
      'iswcs': [],
      'type-id': null,
      'languages': [
        'eng'
      ],
      'id': '52fc9a0f-5b97-4324-80c4-0896d62b1a17',
      'disambiguation': '',
      'attributes': [],
      'type': null,
      'language': 'eng'
    },
    {
      'type': null,
      'attributes': [],
      'disambiguation': '',
      'language': null,
      'id': '541e1da9-5151-31a0-9e45-aab897ac384e',
      'type-id': null,
      'iswcs': [],
      'languages': [],
      'title': 'Days / This Time Tomorrow'
    },
    {
      'type-id': null,
      'iswcs': [],
      'languages': [
        'eng'
      ],
      'id': '60af798f-c9b0-4f0c-a787-1cc3a4a73bd1',
      'attributes': [],
      'disambiguation': '',
      'type': null,
      'language': 'eng',
      'title': 'Cold Arms'
    },
    {
      'type-id': null,
      'iswcs': [
        'T-902.772.089-4'
      ],
      'languages': [
        'eng'
      ],
      'id': '642c1f9b-e0f7-45aa-8df6-3a9d2f261013',
      'disambiguation': '',
      'attributes': [],
      'type': null,
      'language': 'eng',
      'title': 'But My Heart Told My Head'
    },
    {
      'title': '42',
      'disambiguation': '',
      'attributes': [],
      'type': null,
      'language': 'eng',
      'iswcs': [],
      'type-id': null,
      'languages': [
        'eng'
      ],
      'id': '71d45c2b-fd53-4d7d-abb2-33eb517edca6'
    },
    {
      'id': '75ee330f-6206-38f5-be76-3033628308bf',
      'languages': [
        'eng'
      ],
      'type-id': null,
      'iswcs': [],
      'type': null,
      'disambiguation': '',
      'attributes': [],
      'language': 'eng',
      'title': 'Friend of the Devil'
    },
    {
      'attributes': [],
      'disambiguation': '',
      'type': null,
      'language': 'eng',
      'languages': [
        'eng'
      ],
      'type-id': null,
      'iswcs': [],
      'id': '7649c4cb-81b3-4cad-8c06-cb917809e3e2',
      'title': 'Broad-Shouldered Beasts'
    },
    {
      'title': 'Hold On to What You Believe',
      'id': '8348a018-7668-4bb2-96c6-603ffcb0c575',
      'languages': [
        'eng'
      ],
      'type-id': null,
      'iswcs': [
        'T-905.683.078-5'
      ],
      'language': 'eng',
      'type': null,
      'disambiguation': '',
      'attributes': []
    },
    {
      'language': 'eng',
      'disambiguation': '',
      'attributes': [],
      'type': null,
      'languages': [
        'eng'
      ],
      'type-id': null,
      'iswcs': [],
      'id': '8b75b51e-26fb-4d9e-868a-7aa58f0eea1f',
      'title': 'Believe'
    },
    {
      'title': 'Guiding Light',
      'attributes': [],
      'disambiguation': '',
      'type': null,
      'language': 'eng',
      'languages': [
        'eng'
      ],
      'type-id': null,
      'iswcs': [],
      'id': '8c1212bc-178a-43c9-9b5a-2b8d32cbd5cc'
    },
    {
      'id': '93bf7782-5761-4706-a24f-b861b9fb7b81',
      'iswcs': [
        'T-912.678.390-2'
      ],
      'type-id': null,
      'languages': [
        'eng'
      ],
      'language': 'eng',
      'type': null,
      'disambiguation': '',
      'attributes': [],
      'title': 'Broken Crown'
    },
    {
      'id': '9618c66d-3390-43b4-b500-d10c6d6169cf',
      'type-id': null,
      'languages': [
        'eng'
      ],
      'iswcs': [
        'T-909.234.730-3'
      ],
      'type': null,
      'disambiguation': '',
      'attributes': [],
      'language': 'eng',
      'title': 'Below My Feet'
    },
    {
      'title': 'Feel The Tide',
      'attributes': [],
      'disambiguation': '',
      'type': null,
      'language': 'eng',
      'iswcs': [
        'T-901.662.477-4'
      ],
      'type-id': null,
      'languages': [
        'eng'
      ],
      'id': 'c7a0a34f-b55f-4b0c-887c-d9301710900c'
    },
    {
      'title': 'Forever',
      'languages': [
        'eng'
      ],
      'type-id': null,
      'iswcs': [],
      'id': 'ce0fcfda-9043-4f9f-aa52-d979f721cc14',
      'language': 'eng',
      'disambiguation': '',
      'attributes': [],
      'type': null
    },
    {
      'title': 'Awake My Soul',
      'type': null,
      'disambiguation': '',
      'attributes': [],
      'language': 'eng',
      'id': 'd0b77c4a-d934-4cbb-b858-07384679902e',
      'languages': [
        'eng'
      ],
      'type-id': null,
      'iswcs': [
        'T-901.662.476-3'
      ]
    },
    {
      'language': 'eng',
      'type': null,
      'attributes': [],
      'disambiguation': '',
      'id': 'dd58fc16-8118-40fc-8fb3-4dd3aaf66f83',
      'languages': [
        'eng'
      ],
      'type-id': null,
      'iswcs': [
        'T-909.234.738-1'
      ],
      'title': 'Hopeless Wanderer'
    },
    {
      'type-id': null,
      'languages': [
        'eng'
      ],
      'iswcs': [],
      'id': 'ece67856-f5ef-4a60-90ab-9b9e537b2e7a',
      'disambiguation': '',
      'attributes': [],
      'type': null,
      'language': 'eng',
      'title': 'Ditmas'
    },
    {
      'language': 'eng',
      'attributes': [],
      'disambiguation': '',
      'type': null,
      'languages': [
        'eng'
      ],
      'type-id': null,
      'iswcs': [
        'T-909.234.734-7'
      ],
      'id': 'eefd964b-4390-4935-af0a-066ac842e9c8',
      'title': 'For Those Below'
    },
    {
      'type-id': null,
      'iswcs': [
        'T-905.465.281-2'
      ],
      'languages': [
        'eng'
      ],
      'id': 'f1730409-e325-4b59-bc10-47e8fad1cb6a',
      'disambiguation': '',
      'attributes': [],
      'type': null,
      'language': 'eng',
      'title': 'Dust Bowl Dance'
    },
    {
      'iswcs': [
        'T-912.678.391-3'
      ],
      'type-id': null,
      'languages': [
        'eng'
      ],
      'id': 'f196f6bb-3def-4d62-98e4-044c39f3ee61',
      'disambiguation': '',
      'attributes': [],
      'type': null,
      'language': 'eng',
      'title': 'Ghosts That We Knew'
    }
  ],
  'country': 'GB',
  'area': {
    'iso-3166-1-codes': [
      'GB'
    ],
    'id': '8a754a16-0027-3a29-b6d7-2b40ea0481ed',
    'sort-name': 'United Kingdom',
    'name': 'United Kingdom',
    'disambiguation': ''
  },
  'ipis': []
};

const lyrics = {
  'lyrics': 'Paroles de la chanson Fix You par Coldplay\nWhen you try your best but you don\'t succeed\n\n\nWhen you get what you want but not what you need\n\n\nWhen you feel so tired but you can\'t sleep \n\n\nStuck in reverse.\n\n\n\n\n\nAnd the tears come streaming down your face \n\n\nWhen you lose something you can\'t replace \n\n\nWhen you love someone but it goes to waste \n\n\nCould it be worse?\n\n\n\n\n\nLights will guide you home \n\n\nAnd ignite your bones\n\n\nAnd I will try to fix you\n\n\n\n\n\nAnd high up above or down below \n\n\nWhen you\'re too in love to let it go \n\n\n\n\n\n\nBut if you never try you\'ll never know \n\n\nJust what you\'re worth\n\n\n\n\n\nLights will guide you home \n\n\nAnd ignite your bones\n\n\nAnd I will try to fix you \n\n\n\n\n\nTears stream, down on your face \n\n\nWhen you lose something you cannot replace \n\n\nTears stream down on your face \n\n\nAnd I..\n\n\n\n\n\nTears stream, down on your face \n\n\nI promise you I will learn from my mistakes \n\n\nTears stream down on your face \n\n\nAnd I..\n\n\n\n\n\nLights will guide you home \n\n\n\n\n\n\nAnd ignite your bones\n\n\nAnd I will try to fix you.'
};

export const getArtists = () => {
  return Promise.resolve({
    artists,
    count: 44,
    offset: 0
  });
};

export const getArtist = () => {
  return Promise.resolve({
    artist,
    works: artist.works
  });

};

export const getLyrics = () => {
  return Promise.resolve({
    lyrics
  }); 
};
