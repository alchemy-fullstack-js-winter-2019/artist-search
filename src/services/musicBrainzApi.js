export const getArtists = (artist, offset) => {
  return fetch(`http://musicbrainz.org/ws/2/artist?query=${artist}&fmt=json&limit=25&offset=${offset}`)
    .then(res => res.json())
    .then(json => ({
      artists: json.artists,
      count: json.count,
      offset: json.offset
    }));
};

export const getArtist = (id) => {
  return fetch(`http://musicbrainz.org/ws/2/artist/${id}?fmt=json&inc=works`)
    .then(res =>res.json())
    .then(json => ({
      artist: json.name,
      works: json.works
    }));
};

export const getLyrics = (artist, work) => {
  console.log(work);
  return fetch(`https://api.lyrics.ovh/v1/${artist}/${work}`)
    .then(res => res.json())
    .then(json => ({
      lyrics: json.lyrics
    })); 

};
