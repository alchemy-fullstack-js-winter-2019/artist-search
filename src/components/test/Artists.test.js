import React from 'react';
import Artists from '../Artists';
import renderer from 'react-test-renderer';
import { getArtists } from '../../services/musicBrainzApi';

jest.mock('../../services/musicBrainzApi.js');

describe('Artists', () => {
  it('matches a snapshot', () => {
    const artists = [
      {
        'id': '57f73e51-5b95-485b-9102-bc7bd23584f7',
        'type': 'Group',
        'type-id': 'e431f5f6-b5d2-343d-8b36-72607fffb74b',
        'score': 100,
        'name': 'Mumford’s',
        'sort-name': 'Mumford’s',
        'life-span': {
          'begin': '2008',
          'ended': null
        },
        'tags': [
          {
            'count': 1,
            'name': 'iowa'
          }
        ]
      },
      {
        'id': 'e5379ae1-cb4c-46dc-89f4-3d22791f589a',
        'score': 86,
        'name': 'Quincy Mumford',
        'sort-name': 'Mumford, Quincy',
        'life-span': {
          'ended': null
        }
      },
      {
        'id': '2f3613cf-568c-4e18-ba59-ce6d50f05f00',
        'type': 'Group',
        'type-id': 'e431f5f6-b5d2-343d-8b36-72607fffb74b',
        'score': 85,
        'name': 'Brock Mumford',
        'sort-name': 'Brock Mumford',
        'life-span': {
          'ended': null
        }
      }];
    const tree = renderer.create(
      <Artists artists={artists} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
