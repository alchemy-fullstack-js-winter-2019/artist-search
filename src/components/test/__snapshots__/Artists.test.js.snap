// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists matches a snapshot 1`] = `
<ul
  className="artists"
>
  <a
    href="/artist/57f73e51-5b95-485b-9102-bc7bd23584f7"
    onClick={[Function]}
  >
    <li>
      Mumfordâ€™s
    </li>
  </a>
  <a
    href="/artist/e5379ae1-cb4c-46dc-89f4-3d22791f589a"
    onClick={[Function]}
  >
    <li>
      Quincy Mumford
    </li>
  </a>
  <a
    href="/artist/2f3613cf-568c-4e18-ba59-ce6d50f05f00"
    onClick={[Function]}
  >
    <li>
      Brock Mumford
    </li>
  </a>
</ul>
`;
